spring-integration-core/src/main/java/org/springframework/integration/transaction/IntegrationResourceHolderSynchronization.java

```diff
@@ -24,11 +24,13 @@
  * @author Artem Bilan
  * @since 4.0
  */
-public abstract class IntegrationResourceHolderSynchronization
+public class IntegrationResourceHolderSynchronization
 		extends ResourceHolderSynchronization<IntegrationResourceHolder, Object> {
 
 	protected final IntegrationResourceHolder resourceHolder;
 
+	private boolean shouldUnbindAtCompletion;
```

Comment:
`true` by default.
That is only `AbstractPollingEndpoint` to take care about the `TL` by its own.